gobblin-utility/src/main/java/gobblin/util/io/HttpInputStream.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+
+
+/**
+ * An extension of {@link FSDataInputStream} that takes a {@link HttpURLConnection} and closes the connection in {@link #close()} method.
+ */
+public class HttpInputStream extends FSDataInputStream {
```

Comment:
This could be more generic, taking any `Closeable` or even a set of `Closeables` that will al be closed at the same time as the `InputStream`.